cmd/httpd/handlers/translations.go

```diff
@@ -0,0 +1,210 @@
+		return
+	}
+
+	options := data.FindOptions{
+		Filter: map[string]interface{}{},
+		Limit:  pagination.Size,
+		Skip:   pagination.Number - 1,
+	}
+
+	list, error := h.TranslationHelper.Find(r.Context(), options)
+
+	if error != nil {
+		web.RespondWithInternal(r.Context(), w)
+		return
+	}
+
+	response.Data = list
+	response.Links = links
+	response.Meta = map[string]interface{}{
+		"totalPages": math.Ceil(float64(count) / float64(pagination.Size)),
+	}
+
+	web.RespondWith(r.Context(), w, http.StatusOK, response)
+}
+
+func (h *translations) create(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var newTranslation data.NewTranslation
+	var response = web.Response{}
+
+	errors := web.Decode(r, &newTranslation)
+
+	if len(errors) > 0 {
+		response.Errors = errors
+		web.RespondWith(r.Context(), w, http.StatusBadRequest, response)
+		return
+	}
+
+	translation, error := h.TranslationHelper.New(r.Context(), newTranslation)
+
+	if error != nil {
+		web.RespondWithInternal(r.Context(), w)
+		return
+	}
+
+	response.Data = translation
+
+	web.RespondWith(r.Context(), w, http.StatusOK, response)
+}
+
+func (h *translations) get(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var response = web.Response{}
+	var word = ps.ByName("id")
+
+	sourceLanguageID, exist := r.URL.Query()["source"]
+	targetLanguageID, exist := r.URL.Query()["target"]
+
+	if !exist || len(sourceLanguageID) == 0 {
+		response.Errors = web.Error{
+			Status: 400,
+			Source: web.ErrorSource{
+				Pointer: "source language is missing",
+			},
+			Detail: "Missing values",
+		}
+		web.RespondWith(r.Context(), w, http.StatusBadRequest, response)
+		return
+	}
+
+	if !exist || len(targetLanguageID) == 0 {
+		response.Errors = web.Error{
+			Status: 400,
+			Source: web.ErrorSource{
+				Pointer: "target language is missing",
+			},
+			Detail: "Missing values",
+		}
+		web.RespondWith(r.Context(), w, http.StatusBadRequest, response)
+		return
+	}
+
+	ID, error := primitive.ObjectIDFromHex(word)
+
+	if error != nil {
+		web.RespondWithBadRequest(r.Context(), w)
+		return
+	}
+
+	sourceWord, error := h.WordsHelper.FindByID(r.Context(), sourceLanguageID[0], ID)
+
+	if error != nil {
+		web.RespondWithNotFound(r.Context(), w)
+		return
+	}
+
+	targetWordID, error := h.TranslationHelper.FindByID(r.Context(), sourceWord.ID, sourceLanguageID[0], targetLanguageID[0])
+
+	if error != nil {
+		web.RespondWithNotFound(r.Context(), w)
+		return
+	}
+
+	targetWord, error := h.WordsHelper.FindByID(r.Context(), targetLanguageID[0], *targetWordID)
+
+	if error != nil {
+		web.RespondWithNotFound(r.Context(), w)
+		return
+	}
+
+	targetLanguage, error := h.LanguagesHelper.FindByID(r.Context(), targetLanguageID[0])
+
+	if error != nil {
+		web.RespondWithBadRequest(r.Context(), w)
+		return
+	}
+
+	traduction := Traduction{
+		SourceLanguage: sourceLanguageID[0],
+		SourceWord:     sourceWord.ID,
+		TargetLanguage: targetLanguage.ID,
```

Comment:
Only to clarify here, the value that we use around the project is the `collection_name` and not the _id provided by `MongoDB`, this property id matches with the ObjectID.

Why?

Because once you have the language code (collection_name) you will not need to do a find to get the collection for an specific language, the language name itself will be the collection name.

Reply:
But, if you think it's better to use the ID instead of the collection_name, let me know. :+1:  Maybe I'm not seeing something here.

Because for the words resource, i'm ussing the collection_name instead of the language `_id`. 

If we will change the usage of the `_id` to identify a language, maybe would be needed change that crud as well to keep a standard in the code.

I'll appreciate your comments here.

Reply:
Yes, and that ID is not the `MongoDB` _id.

I am using the `collection_name `on that but in the struct of languages, the property name to `collection_name `is `ID`, so don't worry, it works like you say.

Reply:
:ok_hand: